### PR TITLE
add component definition

### LINKED ISSUE #13: Component guide
Nice abstraction of original guide to ES6 format.  

How about including a section for Angular Components (those introduced in 1.5)

I have a nice abstraction convention myself, but would be curious how it compares to others.

### PR DESCRIPTION
**Pull request by mikeerickson**
@rwwagner90 

OK, here is the updated definition, including component definition.  I feel some more clean up and definition needs to be done before merging, but wanted to get something to you so you can have a look.

Resolves #13

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,28 +63,13 @@ it is expected that the imported class was defined correctly, in another file, a
       .factory('someFactory', someFactory);
   ```
 
-  The same components are now separated into their own files.
+  The same directives are now separated into their own files.
 
-  ```js
```
Comment (RobbieTheWagner) [inline README.md:67]: Why are we deleting this whole example?

Comment (mikeerickson) [inline README.md:67]: Most likely an error on my part.  Let me have a look and see what happened.

File: README.md [hunk 2]
```diff
@@ -1358,19 +1392,30 @@ controllerAs can also be used in the router like so:
     app/
         app.module.js
         app.config.js
-        components/
-            calendar.directive.js
-            calendar.directive.html
-            user-profile.directive.js
-            user-profile.directive.html
+        compoments/
```
Comment (RobbieTheWagner) [inline README.md]: typo

File: README.md [hunk 3]
```diff
@@ -1358,19 +1392,30 @@ controllerAs can also be used in the router like so:
...
+            compliment/
+              Compliment.component.js
```
Comment (RobbieTheWagner) [inline README.md]: Shouldn't this be lowercase compliment?

File: README.md (hunk 1 above)
Comment (mikeerickson) [inline README.md:67]: Something looks awry here... Diff is showing a lot more edits than I recall making?  I will clean this up and push up a revised document...

Comment (RobbieTheWagner) [inline README.md:67]: I think a lot of it was just stripping spaces out from differences in text editors, not sure though.

Comment (mikeerickson) [inline README.md:67]: Yes, that is what I am seeing as well.  But this should not be happening?  I will have to review my `.gitattributes` settings.

Comment (RobbieTheWagner) [inline README.md:67]: I think I may have always edited directly in github, not in an editor

File: README.md [hunk 4]
```diff
@@ -1052,9 +1102,9 @@ controllerAs can also be used in the router like so:
 ### Feature File Names
 ###### [Style [Y121](#style-y121)]
 
-  - Use consistent names for all components following a pattern that describes the component's feature then (optionally) its type. My recommended pattern is `feature.type.js`.
+  - Use consistent names for all parts following a pattern that describes the parts feature then (optionally) its type. My recommended pattern is `feature.type.js`.
 
-    *Why?*: Provides a consistent way to quickly identify components.
+    *Why?*: Provides a consistent way to quickly identify individula parts.
```
Comment (RobbieTheWagner) [inline README.md]: typo

**Comment by RobbieTheWagner** [comment]
@mikeerickson I'm not a fan of the word parts being switched out for components. Could we use entity/entities or something else better than parts?

**Comment by mikeerickson** [comment]
@rwwagner90 Ya, that is what I pinged you about this morning (when I sent the email)   

I too am not feeling comfortable about the wording (using parts) but wanted to differentiate the overloaded use of component(s) not to be mistaken with angular "components"   I tried to think of different verbs that could be used (didn't think of entity/entities at the time) but am fine with that)

I will update accordingly and push another update

**Comment by mikeerickson** [comment]
@rwwagner90 All done!

File: README.md [hunk 5]
```diff
@@ -1102,9 +1102,9 @@ angular.module('app')
 ### Feature File Names
 ###### [Style [Y121](#style-y121)]
 
-  - Use consistent names for all parts following a pattern that describes the parts feature then (optionally) its type. My recommended pattern is `feature.type.js`.
+  - Use consistent names for all entities following a pattern that describes the entity feature then (optionally) its type. My recommended pattern is `feature.type.js`.
```
Comment (RobbieTheWagner) [inline README.md]: Should be "entity's" feature

**Comment by RobbieTheWagner** [comment]
@mikeerickson just found one grammatical issue, but looks good otherwise. Are you happy with it and does it get across the points you wanted to make about components? I'd compare with the Todd Motto guide and make sure everything is in good shape, then whenever you feel it's good to go we can merge.

**Comment by mikeerickson** [comment]
@rwwagner90 I am proofing this one more time (made the fix)   Also, my statements are loosely based on those from Todd's guide so the answer is yes :-)

Update coming soon!

**Comment by mikeerickson** [comment]
@rwwagner90 all done!

File: README.md [hunk 6]
```diff
@@ -1102,7 +1104,7 @@ angular.module('app')
 ### Feature File Names
 ###### [Style [Y121](#style-y121)]
 
-  - Use consistent names for all entities following a pattern that describes the entity feature then (optionally) its type. My recommended pattern is `feature.type.js`.
+  - Use consistent names for all entities following a pattern that describes the entities feature then (optionally) its type. My recommended pattern is `feature.type.js`.
```
Comment (RobbieTheWagner) [inline README.md]: entity's

**Comment by RobbieTheWagner** [comment]
@mikeerickson still grammatical issues. Used entities where it should be entity's

**Comment by mikeerickson** [comment]
@rwwagner90 Full disclosure, English was a rough subject for me in school.   I was a jock and felt I would never need to use it, ya I am proving that now huh.

Updated.

**Comment by RobbieTheWagner** [comment]
@mikeerickson haha no worries. Thanks for making the updates!

**Comment by RobbieTheWagner** [comment]
Merged!

**Comment by mikeerickson** [comment]
@rwwagner90 Excellent, thanks!  Have a nice weekend :-)

**Pull request merged**